### ISSUE TITLE
1.15.0 regression: `list.append` false positive with `Signals` enum

### ISSUE THREAD

**Issue by sobolevn** [python/mypy#18628]
This code:

```python
import signal
signals = []
if hasattr(signal, 'SIGALRM'):
    signals.append(signal.SIGALRM)
if hasattr(signal, 'SIGUSR1'):
    signals.append(signal.SIGUSR1)
```

Produces:

```
ex.py:7: error: Argument 1 to "append" of "list" has incompatible type "Literal[Signals.SIGUSR1]"; expected "Literal[Signals.SIGALRM]"  [arg-type]
```

Source: https://github.com/python/typeshed/blob/main/stdlib/signal.pyi

```python
SIGALRM: Signals
SIGUSR1: Signals
```

It is a regression of 1.15.0, before this code was correct: https://mypy-play.net/?mypy=1.14.1&python=3.12&gist=24f0696ba047856ad7d932f4a7f25ec4
I don't see a reason, why two enum membres cannot be appended to a single list.

**Comment by JukkaL** [comment]
I think this can be fixed by using `SIGALRM: Final = Signals.SIGALRM` etc. in the stubs instead of explicit literal type annotations. Using a literal type annotations instead of `Final` is likely to cause issues like this. The mypy behavior hasn't changed in a long time, so I doubt it's a regression in mypy.

**Comment by sobolevn** [comment]
I confirmed this with the repro:

```python
from typing import Final, Literal
from enum import Enum

class S(Enum):
    A = 1
    B = 2

A1: Literal[S.A]
B1: Literal[S.B]

signals1 = []
signals1.append(A1)
signals1.append(B1)  # E: Argument 1 to "append" of "list" has incompatible type "Literal[S.B]"; expected "Literal[S.A]"

A2: Final = S.A
B2: Final = S.B

signals2 = []
signals2.append(A2)
signals2.append(B2)
```

**Comment by sobolevn** [comment]
I opened https://github.com/python/typeshed/pull/13500

**Comment by sobolevn** [comment]
Typeshed PR is merged. It will be fixed once Typeshed is synced.
